homeassistant/components/brother/config_flow.py

```diff
@@ -64,6 +64,85 @@ class BrotherConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
             step_id="user", data_schema=DATA_SCHEMA, errors=errors
         )
 
+    async def async_step_zeroconf(self, user_input=None):
+        """Handle zeroconf discovery."""
+        if user_input is None:
+            return self.async_abort(reason="connection_error")
+
+        try:
+            if "Brother" not in user_input["name"]:
```

Comment:
this is very weird code. Just do something like this pleaes:

```python
if "Brother" not in user_input.get("name"):
```

Also, do you want to check that it starts with, or is "Two Brothers on the fourth floor" ok too ?